Add plugin: GridExplorer

Devon22 opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/Devon22/obsidian-gridexplorer

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

Devon22 reviewed (commented):
fix

Zachatoo commented:
What differentiates this plugin from other plugins that add a grid view, such as [obsidian-vault-explorer](https://github.com/decaf-dev/obsidian-vault-explorer)?

[const TRANSLATIONS = {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L4-L225)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[const langSetting = window.localStorage.getItem('language');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L229)
Please use [getLanguage](https://docs.obsidian.md/Reference/TypeScript+API/getLanguage) to get the user's current language. Note that you'll also need to update your [minAppVersion](https://docs.obsidian.md/Reference/TypeScript+API/PluginManifest/minAppVersion) to `1.8.0`.

[.button-container {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/styles.css#L192-L230), [.view-content {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/styles.css#L21-L24)
Please don't overwrite core styling in a plugin, add another class/data attribute so it only applies to your plugin.

Please update your CSS classes to be more specific to your plugin. Classes such as `grid-item`, `clear-button`, `header-buttons`, `search-container`, etc are all generic enough that other plugins may unintentionally override your styles, or vice versa.

[bookmarkOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1040-L1044), [searchOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1075-L1079), [backlinksOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1109-L1113), [allNotesOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1139-L1143), [folderOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1179-L1183), [clearButton.style.display = this.defaultQuery ? 'flex' : 'none';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L789), [clearButton.style.display = searchInput.value ? 'flex' : 'none';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L794), [clearButton.style.display = 'none';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L800), [searchButton.style.display = 'none';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L545), [searchText.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L560)
You should avoid assigning styles via JavaScript or in HTML and instead move all these styles into CSS so that they are more easily adaptable by themes and snippets.

[bookmarkOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1055-L1061), [searchOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1090-L1096), [backlinksOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1124-L1130), [allNotesOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1154-L1160), [folderOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1194-L1200)
Use [`:hover`](https://developer.mozilla.org/en-US/docs/Web/CSS/:hover) instead of using JavaScript to apply hover styles.

[containerEl.createEl('h2', { text: t('grid_view_settings') });](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L940)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[const { Modal, TFolder } = require('obsidian');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1019), [const { Modal } = require('obsidian');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L763), [class GridExplorerSettingTab extends require('obsidian').PluginSettingTab {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L930)
Keep all your imports at the top of your .js file.

[this.app.workspace.detachLeavesOfType('grid-view');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1267)
Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: <https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60>

[const folders = app.vault.getAllLoadedFiles()](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1163)
You can use [.getAllFolders()](https://docs.obsidian.md/Reference/TypeScript+API/Vault/getAllFolders) here instead.

[const titleEl = contentArea.createEl('h3');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L618-L619), [const searchText = searchTextContainer.createSpan('search-text');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L557-L558)
Prefer to use add text using the second argument to [`createEl`](https://docs.obsidian.md/Plugins/User+interface/HTML+elements#Create+HTML+elements+using+%60createEl()%60).

[const contentWithoutFrontmatter = content.replace(...);](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L667)
Prefer to use [getFrontMatterInfo](https://docs.obsidian.md/Reference/TypeScript+API/getFrontMatterInfo) to get a note's content without frontmatter.

[await this.app.vault.trash(file, false);](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L749)
Prefer to use [`FileManager.trashFile()`](https://docs.obsidian.md/Reference/TypeScript+API/FileManager/trashFile) to remove a file or folder according to the user's preferred 'trash' options.

---

To make the manual review process and future plugin maintenance more efficient, we recommend that you split your code into multiple files and move all source code files into a `src` directory. You can use a bundling tool such as `esbuild` to bundle your code into a single `main.js` file for releases. See [obsidian-sample-plugin](https://github.com/obsidianmd/obsidian-sample-plugin) for an example.

Devon22 commented:
> What differentiates this plugin from other plugins that add a grid view, such as [obsidian-vault-explorer](https://github.com/decaf-dev/obsidian-vault-explorer)?

Compared to obsidian-vault-explorer, GridExplorer is more like a file browser. You can freely switch between folders and browse files, just like using File Explorer on your computer. Other plugins might focus more on the relationships between notes, but I want to provide a more intuitive way to manage files.

> [const TRANSLATIONS = {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L4-L225) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)
> 
> [const langSetting = window.localStorage.getItem('language');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L229) Please use [getLanguage](https://docs.obsidian.md/Reference/TypeScript+API/getLanguage) to get the user's current language. Note that you'll also need to update your [minAppVersion](https://docs.obsidian.md/Reference/TypeScript+API/PluginManifest/minAppVersion) to `1.8.0`.
> 
> [.button-container {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/styles.css#L192-L230), [.view-content {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/styles.css#L21-L24) Please don't overwrite core styling in a plugin, add another class/data attribute so it only applies to your plugin.
> 
> Please update your CSS classes to be more specific to your plugin. Classes such as `grid-item`, `clear-button`, `header-buttons`, `search-container`, etc are all generic enough that other plugins may unintentionally override your styles, or vice versa.
> 
> [bookmarkOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1040-L1044), [searchOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1075-L1079), [backlinksOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1109-L1113), [allNotesOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1139-L1143), [folderOption.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1179-L1183), [clearButton.style.display = this.defaultQuery ? 'flex' : 'none';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L789), [clearButton.style.display = searchInput.value ? 'flex' : 'none';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L794), [clearButton.style.display = 'none';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L800), [searchButton.style.display = 'none';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L545), [searchText.style.cursor = 'pointer';](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L560) You should avoid assigning styles via JavaScript or in HTML and instead move all these styles into CSS so that they are more easily adaptable by themes and snippets.
> 
> [bookmarkOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1055-L1061), [searchOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1090-L1096), [backlinksOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1124-L1130), [allNotesOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1154-L1160), [folderOption.addEventListener('mouseenter', () => {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1194-L1200) Use [`:hover`](https://developer.mozilla.org/en-US/docs/Web/CSS/:hover) instead of using JavaScript to apply hover styles.
> 
> [containerEl.createEl('h2', { text: t('grid_view_settings') });](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L940) [Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)
> 
> [const { Modal, TFolder } = require('obsidian');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1019), [const { Modal } = require('obsidian');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L763), [class GridExplorerSettingTab extends require('obsidian').PluginSettingTab {](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L930) Keep all your imports at the top of your .js file.
> 
> [this.app.workspace.detachLeavesOfType('grid-view');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1267) Do not detach leaves with your custom view in `onunload`, this is an antipattern, see: https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Don't+detach+leaves+in+%60onunload%60
> 
> [const folders = app.vault.getAllLoadedFiles()](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L1163) You can use [.getAllFolders()](https://docs.obsidian.md/Reference/TypeScript+API/Vault/getAllFolders) here instead.
> 
> [const titleEl = contentArea.createEl('h3');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L618-L619), [const searchText = searchTextContainer.createSpan('search-text');](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L557-L558) Prefer to use add text using the second argument to [`createEl`](https://docs.obsidian.md/Plugins/User+interface/HTML+elements#Create+HTML+elements+using+%60createEl()%60).
> 
> [const contentWithoutFrontmatter = content.replace(...);](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L667) Prefer to use [getFrontMatterInfo](https://docs.obsidian.md/Reference/TypeScript+API/getFrontMatterInfo) to get a note's content without frontmatter.
> 
> [await this.app.vault.trash(file, false);](https://github.com/Devon22/obsidian-gridexplorer/blob/95270a5d9c276228432c876d360913f1293a512c/main_src.js#L749) Prefer to use [`FileManager.trashFile()`](https://docs.obsidian.md/Reference/TypeScript+API/FileManager/trashFile) to remove a file or folder according to the user's preferred 'trash' options.
> 
> To make the manual review process and future plugin maintenance more efficient, we recommend that you split your code into multiple files and move all source code files into a `src` directory. You can use a bundling tool such as `esbuild` to bundle your code into a single `main.js` file for releases. See [obsidian-sample-plugin](https://github.com/obsidianmd/obsidian-sample-plugin) for an example.

Fixed.

Zachatoo commented:
Another note, when you do decide to add setting headings, make sure you do it like this

```ts
new Setting(containerEl).setName('name here').setHeading();
```

Looks like the bot thinks you didn't make all the required changes, I'll check back on this in a few days.

Devon22 commented:
> Another note, when you do decide to add setting headings, make sure you do it like this
> 
> ```ts
> new Setting(containerEl).setName('name here').setHeading();
> ```
> 
> Looks like the bot thinks you didn't make all the required changes, I'll check back on this in a few days.

I have split my code into multiple files and moved all source code files into a `src` directory as recommended. I would appreciate it if you could review the updated code to ensure everything is in order.

joethei commented:
["main": "main.js",](https://github.com/Devon22/obsidian-gridexplorer/blob/6d24c8a8232dce585d51ca7da6c4b411547adf0c/manifest.json#L7)
This is not a thing, remove it.

['grid_view_title': 'Grid View',](https://github.com/Devon22/obsidian-gridexplorer/blob/6d24c8a8232dce585d51ca7da6c4b411547adf0c/src/translations.ts#L96), ['open_grid_view': 'Open Grid View',](https://github.com/Devon22/obsidian-gridexplorer/blob/6d24c8a8232dce585d51ca7da6c4b411547adf0c/src/translations.ts#L128)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

Devon22 commented:
> ["main": "main.js",](https://github.com/Devon22/obsidian-gridexplorer/blob/6d24c8a8232dce585d51ca7da6c4b411547adf0c/manifest.json#L7) This is not a thing, remove it.
> 
> ['grid_view_title': 'Grid View',](https://github.com/Devon22/obsidian-gridexplorer/blob/6d24c8a8232dce585d51ca7da6c4b411547adf0c/src/translations.ts#L96), ['open_grid_view': 'Open Grid View',](https://github.com/Devon22/obsidian-gridexplorer/blob/6d24c8a8232dce585d51ca7da6c4b411547adf0c/src/translations.ts#L128) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

Fixed.

Devon22 commented:
`Validate Plugin Entry / plugin-validation (pull_request_target)Failing`

How can I handle this error?
